test_firebase.py

```diff
@@ -0,0 +1,40 @@
+"""Tests for firebase.py"""
+from unittest.mock import patch
+from absl.testing import absltest # pylint: disable=no-name-in-module
+from firebase_admin import auth
+from mockfirestore import MockFirestore
+from utilities import firebase
+
+class TestFirebase(absltest.TestCase):
+    """Firebase Testing Class"""
+    decoded_jwt = {
+        'uid': 'f00',
+    }
+
+    @classmethod
+    def setUp(cls): # pylint: disable=invalid-name
+        """Set-up
+            * Mock Firestore database
+        """
+        firebase.db = MockFirestore()
+
+    def test_logged_in(self):
+        """Test checking if a user is logged in"""
+        with patch('utilities.firebase.auth.verify_id_token') as mocked_verify_id_token:
+            mocked_verify_id_token.return_value = self.decoded_jwt
+            logged_in = firebase.logged_in('good_idT0ken')
+            self.assertTrue(logged_in)
+            mocked_verify_id_token.side_effect = \
```

Comment:
this should be two tests

generally one unit test consists of **arrange**, **act**, **assert**:
1. arrange: prepare data, mock stuff, etc.
2. act: make the api call or function call.
3. assert: check that **act** did what it was supposed to, e.g. returned a response with status code 0 and body contents etc.

this test has:
1. arrange (patch and mock)
2. act (call logged_in)
3. assert (assertTrue)
4. arrange (side_effect =)
5. act (call logged_in)
6. assert (assertFalse)

Reply:
Thanks for calling this out! I agree and felt that while working on this but just put them both in one since I saw an example doing it (not a good reason 😅).